dist/gcc-compatible/Hacl_Streaming_HMAC.c

```diff
@@ -0,0 +1,2204 @@
+        uint64_t *buf = (uint64_t *)KRML_HOST_CALLOC(25U, sizeof (uint64_t));
+        s = ((Hacl_Agile_Hash_state_s){ .tag = SHA3_512_s, { .case_SHA3_512_s = buf } });
+        break;
+      }
+    case Hacl_Agile_Hash_Blake2S_32:
+      {
+        uint32_t *buf = (uint32_t *)KRML_HOST_CALLOC(16U, sizeof (uint32_t));
+        s = ((Hacl_Agile_Hash_state_s){ .tag = Blake2S_s, { .case_Blake2S_s = buf } });
+        break;
+      }
+    case Hacl_Agile_Hash_Blake2S_128:
+      {
+        #if HACL_CAN_COMPILE_VEC128
+        s =
+          (
+            (Hacl_Agile_Hash_state_s){
+              .tag = Blake2S_128_s,
+              { .case_Blake2S_128_s = Hacl_Hash_Blake2s_Simd128_malloc_with_key() }
+            }
+          );
+        #else
+        s =
+          KRML_EABORT(Hacl_Agile_Hash_state_s,
+            "provably unreachable code: did an unverified caller violate a precondition\?");
+        #endif
+        break;
+      }
+    case Hacl_Agile_Hash_Blake2B_32:
+      {
+        uint64_t *buf = (uint64_t *)KRML_HOST_CALLOC(16U, sizeof (uint64_t));
+        s = ((Hacl_Agile_Hash_state_s){ .tag = Blake2B_s, { .case_Blake2B_s = buf } });
+        break;
+      }
+    case Hacl_Agile_Hash_Blake2B_256:
+      {
+        #if HACL_CAN_COMPILE_VEC256
+        s =
+          (
+            (Hacl_Agile_Hash_state_s){
+              .tag = Blake2B_256_s,
+              { .case_Blake2B_256_s = Hacl_Hash_Blake2b_Simd256_malloc_with_key() }
+            }
+          );
+        #else
+        s =
+          KRML_EABORT(Hacl_Agile_Hash_state_s,
+            "provably unreachable code: did an unverified caller violate a precondition\?");
+        #endif
+        break;
+      }
+    default:
+      {
+        KRML_HOST_EPRINTF("KaRaMeL incomplete match at %s:%d\n", __FILE__, __LINE__);
+        KRML_HOST_EXIT(253U);
+      }
+  }
+  Hacl_Agile_Hash_state_s
+  *buf = (Hacl_Agile_Hash_state_s *)KRML_HOST_MALLOC(sizeof (Hacl_Agile_Hash_state_s));
+  buf[0U] = s;
+  return buf;
+}
+
+static void init(Hacl_Agile_Hash_state_s *s)
+{
+  Hacl_Agile_Hash_state_s scrut = *s;
+  if (scrut.tag == MD5_s)
+  {
+    uint32_t *p1 = scrut.case_MD5_s;
+    Hacl_Hash_MD5_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA1_s)
+  {
+    uint32_t *p1 = scrut.case_SHA1_s;
+    Hacl_Hash_SHA1_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA2_224_s)
+  {
+    uint32_t *p1 = scrut.case_SHA2_224_s;
+    Hacl_Hash_SHA2_sha224_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA2_256_s)
+  {
+    uint32_t *p1 = scrut.case_SHA2_256_s;
+    Hacl_Hash_SHA2_sha256_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA2_384_s)
+  {
+    uint64_t *p1 = scrut.case_SHA2_384_s;
+    Hacl_Hash_SHA2_sha384_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA2_512_s)
+  {
+    uint64_t *p1 = scrut.case_SHA2_512_s;
+    Hacl_Hash_SHA2_sha512_init(p1);
+    return;
+  }
+  if (scrut.tag == SHA3_224_s)
+  {
+    uint64_t *p1 = scrut.case_SHA3_224_s;
+    memset(p1, 0U, 25U * sizeof (uint64_t));
```

Comment:
I can only comment on auto-generated code since I don't know how to read F*, but I would have expected `Hacl_Hash_SHA3_sha224_init` (same for other SHA3 branches) here even if it only does a 0 memset.

Reply:
Same, also worth fixing. Will do.